ipn/ipnlocal/local.go

```diff
@@ -6453,8 +6453,16 @@ func suggestExitNode(report *netcheck.Report, netMap *netmap.NetworkMap, r *rand
 	if report.PreferredDERP == 0 {
 		return res, ErrNoPreferredDERP
 	}
+	var allowedCandidates set.Set[string]
+	if allowed, ok := syspolicy.GetStringArray(syspolicy.AllowedSuggestedExitNodes, []string{}); ok == nil {
+		allowedCandidates = set.SetOf(allowed)
+	}
```

Comment:
Oops I forgot that returned an `error` rather than `bool`.

Use `nil` rather than `[]string{}` for the default so that you can distinguish a policy with 0 entries from the policy not being configured.

```suggestion
	if allowed, err := syspolicy.GetStringArray(syspolicy.AllowedSuggestedExitNodes, nil); err != nil {
		return res, fmt.Errorf("unable to read %s policy: %w", syspolicy.AllowedSuggestedExitNodes, err)
	} else if allowed != nil {
		allowedCandidates = set.SetOf(allowed)
	}
```